providers/amazon/src/airflow/providers/amazon/aws/executors/aws_lambda/lambda_executor.py

```diff
@@ -425,17 +426,27 @@ def process_queue(self, queue_url: str):
                         "Successful Lambda invocation for task %s received from SQS queue.", task_key
                     )
                 else:
-                    # In this case the Lambda likely started but failed at run time since we got a non-zero
-                    # return code. We could consider retrying these tasks within the executor, because this _likely_
-                    # means the Airflow task did not run to completion, however we can't be sure (maybe the
-                    # lambda runtime code has a bug and is returning a non-zero when it actually passed?). So
-                    # perhaps not retrying is the safest option.
                     self.fail(task_key)
-                    self.log.error(
-                        "Lambda invocation for task: %s has failed to run with return code %s",
-                        task_key,
-                        return_code,
-                    )
+                    if queue_url == self.dlq_url and return_code == None:
+                        # DLQ failure: AWS Lambda service could not complete the invocation after retries.
+                        # This indicates a Lambda-level failure (timeout, memory limit, crash, etc.)
+                        # where the function was unable to successfully execute to return a result.
+                        self.log.error(
+                            "Lambda invocation for task: %s failed at the service level (from DLQ). Command: %s",
+                            task_key,
+                            command,
+                        )
+                    else:
+                        # In this case the Lambda likely started but failed at run time since we got a non-zero
+                        # return code. We could consider retrying these tasks within the executor, because this _likely_
+                        # means the Airflow task did not run to completion, however we can't be sure (maybe the
+                        # lambda runtime code has a bug and is returning a non-zero when it actually passed?). So
+                        # perhaps not retrying is the safest option.
+                        self.log.error(
+                            "Lambda invocation for task: %s has failed to run with return code %s",
```

Comment:
We should also probably differentiate this one a bit more as well.

```suggestion
                            "Lambda invocation for task: %s completed but the underlying Airflow task has returned a non-zero exit code %s",
```

I'm also wondering if this branch should be log.debug? Since it's very common for airlfow tasks to fail for a myriad of reasons, and we don't necessarily need a error log in the scheduler logs for each one of those. Thoughts?

Reply:
I definitely see the argument because from the scheduler's perspective it has completed its job and task based failures will have more meaningful logging within their respective logs. 

My only concern would be in situations similar to what I experienced where a task that began as an attempted lambda timeout to cause a catastrophic failure, but was later retried and provided a return code, falling into the second branch and avoiding the DLQ.

Reply:
But I suppose since it is a catchall for a large number of possible failures, log.debug would make sense in nearly all situations.

Reply:
Yeah, that's fair, but in that situation the task still got marked as failed, and that will show up in the Airflow UI, scheduler logs, etc. So users will be more than aware it failed and can investigate and figure out the cause. I'm just not sure the Lambda executor _also_ needs to raise alarms for those failures. Feels redundant and like it most often will just cause excess churn in the logs.

Reply:
I agree, I've made the adjustment in the latest commit, thanks!